pygeoapi/schemas/config/pygeoapi-config-0.x.yml

```diff
@@ -429,6 +429,36 @@ properties:
                                               type: string
                                           minItems: 1
                                           uniqueItems: true
+                                      crs:
+                                           type: array
+                                           description: |-
+                                               supported coordinate reference systems (CRSs).
+                                               pygeoapi will always provide or add the defaults if not specified.
+                                           items:
+                                               type: string
+                                               format: uri
```

Comment:
Does it make sense to add an `enum` property to validate the enumeration of allowed values? Something like

```yaml
items:
    type: string
    format: uri
    enums:
        - http://www.opengis.net/def/crs/OGC/1.3/CRS84
        - http://www.opengis.net/def/crs/EPSG/0/4326
        - etc
```

in this way we can avoid to use unsupported CRSs

Reply:
I think the default should be - http://www.opengis.net/def/crs/OGC/1.3/CRS84 (WGS84 lon, lat ordering) according to the latest [Part 2 Standard](https://docs.opengeospatial.org/is/18-058r1/18-058r1.html) see 6.2.1. For http://www.opengis.net/def/crs/OGC/1.3/CRS84h, impractically, we should inspect the data. In that case users should explicitly configure.

But apart from that with enums we would need to add the entire list (?). The default is now just http://www.opengis.net/def/crs/OGC/1.3/CRS84.

Reply:
yep, adding the entire list of supported CRSes we could help the users to configure and validate explicitly the list defined in the pygeoapi configuration

Reply:
Let me understand: this is a list of hundreds of URIs including all the height-variants like http://www.opengis.net/def/crs/OGC/1.3/CRS84h ? Plus I expect at some point we may want to add regular EPSG: formats.

Reply:
so you think it is too much?

Reply:
Well, I think there are like 22000 projections at least, plus all kinds of variants with height. In `proj` there used to be a file called `epsg`. Can't find the URI list but here an example https://wiki.spatialmanager.com/index.php/Coordinate_Systems_objects_list. 

Like @GeoSander said: we may better migrate the whole pygeoapi config schema-thing to Pydantic.

For now it may be useful to check the CRS-list/array on startup for `(py)proj` support. The 'enum' should really come from there as pyproj will determine if the projection URI is valid/supported. We check already the regular expression for URI in [this util function](https://github.com/geopython/pygeoapi/blob/55ce54a72d38904c87d967e7725eb63cadd3da10/pygeoapi/util.py#L572).

Reply:
I like the migration of pygeoapi config schema to pydantic. @justb4 @GeoSander can we file this improvement in a new issue?

Reply:
Ok, yes, not part of this PR.